### PR TITLE
OWSI-Core 0.11 property service

### PR DESCRIPTION
**Pull request by florianlacreuse**

### CODE REVIEW

File: basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/common/template/MainTemplate.java [hunk 1]
```diff
@@ -233,9 +234,9 @@ protected void onConfigure() {
 		
 		// Footer
 		add(
-				new Label("version", configurer.getVersion())
+				new Label("version", propertyService.get(VERSION))
 						.add(new AttributeModifier("title", new StringResourceModel("common.version.full")
-								.setParameters(configurer.getVersion(), configurer.getOwsiCoreVersion())
+								.setParameters(propertyService.get(VERSION), propertyService.get(OWSI_CORE_VERSION))
```
Comment (yrodiere) [inline basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/common/template/MainTemplate.java]: Shouldn't we use a ApplicationPropertyModel here and on the Label?

File: owsi-core/owsi-core-components/owsi-core-component-spring/src/main/java/fr/openwide/core/spring/property/SpringPropertyIds.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	public static final ImmutablePropertyId<String> OWSI_CORE_VERSION = new ImmutablePropertyId<>("owsi-core.version");
+	public static final ImmutablePropertyId<String> CONFIGURATION_TYPE = new ImmutablePropertyId<>("configurationType");
+	
+	public static final ImmutablePropertyId<Boolean> MIGRATION_LOGGING_MEMORY = new ImmutablePropertyId<>("migration.logging.memory");
+	
+	public static final ImmutablePropertyId<File> TMP_PATH = new ImmutablePropertyId<>("tmp.path");
+	
+	public static final ImmutablePropertyId<Integer> LUCENE_BOOLEAN_QUERY_MAX_CLAUSE_COUNT = new ImmutablePropertyId<>("lucene.booleanQuery.maxClauseCount");
+	
+	public static final ImmutablePropertyId<File> IMAGE_MAGICK_CONVERT_BINARY = new ImmutablePropertyId<>("imageMagick.convertBinary.path");
```
Comment (yrodiere) [inline owsi-core/owsi-core-components/owsi-core-component-spring/src/main/java/fr/openwide/core/spring/property/SpringPropertyIds.java]: Maybe add a _PATH suffix here? As in TMP_PATH?

File: owsi-core/owsi-core-examples/wicket-showcase/wicket-showcase-webapp/src/main/java/fr/openwide/core/showcase/web/application/config/spring/ShowcaseWebappApplicationPropertyRegisterConfig.java [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
 import org.springframework.context.annotation.Configuration;
 
-import fr.openwide.core.jpa.more.business.property.service.IPropertyRegistry;
-import fr.openwide.core.jpa.more.config.spring.AbstractApplicationPropertyRegistryConfig;
-import fr.openwide.core.showcase.web.application.util.property.ShowcaseWebappPropertyIds;
+import fr.openwide.core.spring.config.spring.AbstractApplicationPropertyRegistryConfig;
+import fr.openwide.core.spring.property.service.IPropertyRegistry;
 
 @Configuration
 public class ShowcaseWebappApplicationPropertyRegisterConfig extends AbstractApplicationPropertyRegistryConfig {
```
Comment (yrodiere) [inline owsi-core/owsi-core-examples/wicket-showcase/wicket-showcase-webapp/src/main/java/fr/openwide/core/showcase/web/application/config/spring/ShowcaseWebappApplicationPropertyRegisterConfig.java]: Should be ShowcaseWebappApplicationPropertyRegist**ry**Config

File: basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/config/spring/BasicApplicationWebappApplicationPropertyRegisterConfig.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import static fr.openwide.core.basicapp.web.application.property.BasicApplicationWebappPropertyIds.PORTFOLIO_ITEMS_PER_PAGE;
+import static fr.openwide.core.basicapp.web.application.property.BasicApplicationWebappPropertyIds.PORTFOLIO_ITEMS_PER_PAGE_DESCRIPTION;
+
+import org.springframework.context.annotation.Configuration;
+
+import fr.openwide.core.spring.config.spring.AbstractApplicationPropertyRegistryConfig;
+import fr.openwide.core.spring.property.service.IPropertyRegistry;
+
+@Configuration
+public class BasicApplicationWebappApplicationPropertyRegisterConfig extends AbstractApplicationPropertyRegistryConfig {
```
Comment (yrodiere) [inline basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/config/spring/BasicApplicationWebappApplicationPropertyRegisterConfig.java]: Should be BasicApplicationWebappApplicationPropertyRegist**ry**Config

File: owsi-core/owsi-core-components/owsi-core-component-wicket-more/src/main/java/fr/openwide/core/wicket/more/console/template/ConsoleTemplate.java [hunk 5]
```diff
@@ -117,7 +122,7 @@ protected String load() {
 		add(new BookmarkablePageLink<Void>("logoutLink", LogoutPage.class));
 		
 		// Version
-		add(new Label("version", configurer.getVersion()));
+		add(new Label("version", propertyService.get(VERSION)));
```
Comment (yrodiere) [inline owsi-core/owsi-core-components/owsi-core-component-wicket-more/src/main/java/fr/openwide/core/wicket/more/console/template/ConsoleTemplate.java]: Maybe we could use an ApplicationPropertyModel here.

File: owsi-core/owsi-core-components/owsi-core-component-wicket-more/src/main/java/fr/openwide/core/wicket/more/fileapi/behavior/FileUploadBehavior.java [hunk 6]
```diff
@@ -347,6 +349,17 @@ public void renderHead(Component component, IHeaderResponse response) {
...
+		Component adresse = new WebMarkupContainer("adresse");
+		Component complement = new WebMarkupContainer("complement");
+		int ADRESSE_MAXIMUM_LENGTH = 36;
+		
+		response.render(OnDomReadyHeaderItem.forScript(
+				new JsStatement().$(null).chain(
+						EventsHelper.keyup().chainLabel(),
+						new JsStatement().append("suivant(" + adresse.getMarkupId() + "," + complement.getMarkupId() + "," + ADRESSE_MAXIMUM_LENGTH + ")").render()
+				).render()
+		));
```
Comment (yrodiere) [inline owsi-core/owsi-core-components/owsi-core-component-wicket-more/src/main/java/fr/openwide/core/wicket/more/fileapi/behavior/FileUploadBehavior.java]: Should be removed... :)

**Comment by florianlacreuse** [comment]
Yaonn's comments were taken into account. Thanks for the feedback!

File: basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/common/template/MainTemplate.java (hunk 1 above)
Comment (yrodiere) [inline basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/common/template/MainTemplate.java]: ... and on the label? :)

**Pull request merged**